packages/cli/commands/project/logs.js

```diff
@@ -4,253 +4,154 @@ const {
   ENVIRONMENTS,
 } = require('@hubspot/local-dev-lib/constants/environments');
 const {
-  addAccountOptions,
-  addConfigOptions,
   getAccountId,
   addUseEnvironmentOptions,
 } = require('../../lib/commonOpts');
 const { trackCommandUsage } = require('../../lib/usageTracking');
 const { logger } = require('@hubspot/local-dev-lib/logger');
 // const { outputLogs } = require('../../lib/ui/serverlessFunctionLogs');
-const {
-  fetchProject,
-  fetchDeployComponentsMetadata,
-} = require('@hubspot/local-dev-lib/api/projects');
 const { getTableContents, getTableHeader } = require('../../lib/ui/table');
-// const {
-//   logApiErrorInstance,
-//   ApiErrorContext,
-// } = require('../../lib/errorHandlers/apiErrors');
-// const {
-//   getFunctionLogs,
-//   getLatestFunctionLog,
-//   getProjectAppFunctionLogs,
-//   getLatestProjectAppFunctionLog,
-// } = require('@hubspot/local-dev-lib/api/functions');
+const { logApiErrorInstance } = require('../../lib/errorHandlers/apiErrors');
 
-const { ensureProjectExists } = require('../../lib/projects');
 const { loadAndValidateOptions } = require('../../lib/validation');
 const { uiBetaTag, uiLine, uiLink } = require('../../lib/ui');
 const { projectLogsPrompt } = require('../../lib/prompts/projectsLogsPrompt');
 // const { tailLogs } = require('../../lib/serverlessLogs');
 const { i18n } = require('../../lib/lang');
-// const { EXIT_CODES } = require('../../lib/enums/exitCodes');
+const { EXIT_CODES } = require('../../lib/enums/exitCodes');
+const ProjectLogsManager = require('../../lib/projectLogsManager');
 
 const i18nKey = 'commands.project.subcommands.logs';
 
 const getPrivateAppsUrl = accountId => {
   const baseUrl = getHubSpotWebsiteOrigin(
-    getEnv() === 'qa' ? ENVIRONMENTS.QA : ENVIRONMENTS.PROD
+    getEnv(accountId) === 'qa' ? ENVIRONMENTS.QA : ENVIRONMENTS.PROD
   );
 
   return `${baseUrl}/private-apps/${accountId}`;
 };
 
-// We currently cannot fetch logs directly to the CLI. See internal CLI issue #413 for more information.
-
-// const handleLogsError = (e, name, projectName) => {
-//   if (e.response && e.response.status === 404) {
-//     logger.debug(`Log fetch error: ${e.message}`);
-//     logger.log(i18n(`${i18nKey}.logs.noLogsFound`, { name }));
-//   } else {
-//     logApiErrorInstance(
-//       e,
-//       new ApiErrorContext({ accountId: getAccountId(), projectName })
-//     );
-//     process.exit(EXIT_CODES.ERROR);
-//   }
-// };
-
-// const handleFunctionLog = async (accountId, options) => {
-//   const {
-//     latest,
-//     follow,
-//     compact,
-//     appPath,
-//     functionName,
-//     projectName,
-//   } = options;
-
-//   let logsResp;
+const handleFunctionLog = async (accountId, options) => {
```

Comment:
Clarification: Does this function only return the log "No logs found"? Assuming this is temporary until the BE catches up.

Reply:
Yep, I'll add a clarifying comment

Reply:
I actually ended up deleting it instead https://github.com/HubSpot/hubspot-cli/pull/1131#discussion_r1735044498